test_aodndata/soop/test_soop_xbt_dm.py

```diff
@@ -18,20 +19,26 @@ def setUp(self):
 
     def test_good_netcdf(self):
         handler = self.run_handler(GOOD_NC,
-                                     include_regexes=['IMOS_SOOP-XBT_T_.*\.nc'],
-                                     check_params={'checks': ['cf', 'imos:1.4']}
-                                     )
+                                   include_regexes=['IMOS_SOOP-XBT_T_.*\.nc'],
+                                   check_params={'checks': ['cf', 'imos:1.4']}
+                                   )
         self.assertEqual(len(handler.file_collection), 2)
-        f = handler.file_collection[0]
-        self.assertEqual(f.check_type, PipelineFileCheckType.NC_COMPLIANCE_CHECK)
-        self.assertEqual(f.publish_type, PipelineFilePublishType.HARVEST_UPLOAD)
-        self.assertEqual(f.name, os.path.basename(GOOD_NC))
-        self.assertEqual(f.dest_path,
+        f_nc = handler.file_collection.filter_by_attribute_id('file_type', FileType.NETCDF)[0]
+        self.assertEqual(f_nc.check_type, PipelineFileCheckType.NC_COMPLIANCE_CHECK)
+        self.assertEqual(f_nc.publish_type, PipelineFilePublishType.HARVEST_UPLOAD)
+        self.assertEqual(f_nc.name, os.path.basename(GOOD_NC))
+        self.assertEqual(f_nc.dest_path,
                          'IMOS/SOOP/SOOP-XBT/DELAYED/Line_PX02_Flores-Sea-Torres-Strait/2016/' + os.path.basename(
                              GOOD_NC)
                          )
-        self.assertTrue(f.is_checked)
-        self.assertTrue(f.is_stored)
+        self.assertTrue(f_nc.is_checked)
+        self.assertTrue(f_nc.is_stored)
+
+        f_jpg = handler.file_collection[1]  # cant filter by extension because this is a tempfile generated file until dest_path is set
+        image_path = os.path.join(handler.state_query._storage_broker.prefix,
+                                  f_jpg.dest_path)
```

Comment:
Can't you just access the file locally at `f_jpg.local_path`?

Reply:
no, at this stage it is already uploaded as the handler is creating it.